test/Writer/AbstractTest.php

```diff
@@ -125,6 +128,78 @@ public function testConstructorWithPriorityFilter()
     }
 
     /**
+     * @covers Zend\Log\Writer\AbstractWriter::__construct
+     */
+    public function testConstructorWithFormatterManager()
+    {
+        // Arrange
+        $pluginManager = new FormatterPluginManager(new ServiceManager());
+
+        // Act
+        $writer = new ConcreteWriter([
+            'formatter_manager' => $pluginManager,
+        ]);
+
+        // Assert
+        $this->assertSame($pluginManager, $writer->getFormatterPluginManager());
+    }
+
+    /**
+     * @covers Zend\Log\Writer\AbstractWriter::__construct
+     * @expectedException Zend\Log\Exception\InvalidArgumentException
+     * @expectedExceptionMessage Writer plugin manager must extend Zend\Log\Writer\FormatterPluginManager; received integer
+     */
+    public function testConstructorWithInvalidFormatterManager()
+    {
+        // Arrange
+        // ...
+
+        // Act
+        $writer = new ConcreteWriter([
+            'formatter_manager' => 123,
+        ]);
+
+        // Assert
+        $this->assertInstanceOf(FormatterPluginManager::class, $writer->getFormatterPluginManager());
```

Comment:
Is this assertion even necessary? The exception will be raised during instantiation...